qemu/tests/qemu_guest_agent.py

```diff
@@ -801,27 +801,48 @@ def gagent_check_set_time(self, test, params, env):
         delta = abs(int(guest_time_after) - target_time / 1000000000)
         if delta > 3:
             test.fail("the time set for guest is not the same with target")
-        # Set the system time from the hwclock
-        if params["os_type"] != "windows":
-            move_time_cmd = params["move_time_cmd"]
-            session.cmd("hwclock -w")
-            guest_hwclock_after_set = session.cmd_output("date +%s")
-            error_context.context("hwclock is '%d' " % int(guest_hwclock_after_set),
-                                  logging.info)
-            session.cmd(move_time_cmd)
-            time_after_move = session.cmd_output("date +%s")
-            error_context.context("the time after move back is '%d' "
-                                  % int(time_after_move), logging.info)
-            self.gagent.set_time()
-            guest_time_after_reset = session.cmd_output(get_guest_time_cmd)
-            error_context.context("the time after being reset is '%d' "
-                                  % int(guest_time_after_reset), logging.info)
-            guest_hwclock = session.cmd_output("date +%s")
-            error_context.context("hwclock for compare is '%d' " % int(guest_hwclock),
+
+        # set invalid dometime if needed
+        invalid_time_test = params.get_boolean("invalid_time_test", "no")
+        if invalid_time_test:
```

Comment:
Comparing with this `if this test...if not this test` structure, I will prefer you have two separated function for these two test case since `invalid_time_test` have totally different implementation.